src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -45,6 +45,7 @@ public class ScriptHost : JobHost, IScriptJobHost
         private readonly IDistributedLockManager _distributedLockManager;
         private readonly IFunctionMetadataManager _functionMetadataManager;
         private readonly IHostIdProvider _hostIdProvider;
+        private readonly IHttpRoutesManager _httpRoutesManager;
```

Comment:
I was going to say the layering here is a bit off, since ScriptHost really shouldn't know about eternal http routes, however, I see other places in this file where we're special casing http functions...

Reply:
Yeah, I agree. That's why I had initially gone with the hosted service approach, keeping that isolated. Not a big fan of how this ended up, but have some plans for cleaning some of this (and other logic here) up a bit.